webapp/components/survey/NodeDefDetails/store/actions/useSetNodeDefProp.js

```diff
@@ -25,48 +27,47 @@ const _checkCanChangeProp = (dispatch, nodeDef, key, value) => {
   return true
 }
 
-export const useSetNodeDefProp = ({ nodeDefState, setNodeDefState }) => (key, value = null, advanced = false) => async (
-  dispatch,
-  getState
-) => {
-  const state = getState()
-  let survey = SurveyState.getSurvey(state)
-  const surveyCycleKey = SurveyState.getSurveyCycleKey(state)
+export const useSetProp = ({ nodeDefState, setNodeDefState }) => {
+  const dispatch = useDispatch()
+  let survey = useSelector(SurveyState.getSurvey)
+  const surveyCycleKey = useSelector(SurveyState.getSurveyCycleKey)
+  const validateNodeDef = useValidateNodeDef({ nodeDefState, setNodeDefState })
 
-  const nodeDef = NodeDefState.getNodeDef(nodeDefState)
+  return (key, value = null, advanced = false) => {
```

Comment:
better to use object param {key, value, advanced}

Reply:
done